packages/taxdat/R/setup_helpers.R

```diff
@@ -1,3 +1,164 @@
+                        "water_access", "open_defe",            
+                        "stunting", "wasting", 
+                        "access_cities", "access_healthcare_motorized", 
+                        "access_healthcare_walking")
+  }else{
+    all_covariates <- names(yaml::read_yaml(covariate_list)) #tmp default: "Layers/middle-distance_covariates.yml"
+  }
+
+  ### Make the check list 
+  check_list <- list(
+    name = "no-check", 
+    countries = "no-check", 
+    countries_name = "no-check", 
+    country_data_report_filename = function(nm, cty, ctynm, param){
+                if(is.null(param)){return(paste(nm, cty, ctynm, "country-data-report.html", sep="_"))}
+                return(param)
+              }, 
+    aoi = function(param){
+        if(cholera_testing){return(param)}
+        return("raw")
+      }, 
+    res_space = "no-check", 
+    res_time = as.function(check_time_res), 
+    grid_rand_effects_N = as.function(check_grid_rand_effects_N), 
+    case_definition = as.function(check_case_definition), 
+    start_time = function(param){
+                if(is.null(param)){stop("The start_time parameter should not be blank because there is no default")}
+                return(param)
+              },   
+    end_time = function(param){
+              if(is.null(param)){stop("The end_time parameter should not be blank because there is no default")}
+              return(param)
+            }, 
+    data_source = function(param){
+                if(is.null(param)){return("sql")}
+                return(param)
+              }, 
+    ovrt_metadata_table = function(param){
+                        if(is.null(param)){return("no")}
+                        return(param)
+                      }, 
+    OCs = "no-check", 
+    taxonomy = function(param){
+              if(is.null(param)){return("taxonomy-working/working-entry1")}
+              return(param)
+            },  
+    covariate_choices = as.function(check_covariate_choices), 
+    obs_model = function(param){
+              if(is.null(param)){return(1)}
+              if(!as.numeric(param) %in% 1:3){return(1)}
+              return(as.numeric(param))
+            }, 
+    od_param = function(param1, param2){
+              if(as.numeric(param1) == 2 | as.numeric(param1) == 3){return(as.numeric(param2))}
+              return(NULL)
+            }, 
+    time_effect = "stan-check", 
+    time_effect_autocorr = "stan-check", 
+    use_intercept = "stan-check", 
+    covariate_transformations = "no-check", 
+    beta_sigma_scale = "stan-check", 
+    sigma_eta_scale = "stan-check", 
+    exp_prior = "stan-check", 
+    do_infer_sd_eta = "stan-check", 
+    do_zerosum_cnst = "stan-check", 
+    use_weights = "stan-check", 
+    covar_warmup = "stan-check", 
+    warmup = "stan-check", 
+    aggregate = as.function(check_aggregate), 
+    tfrac_thresh = as.function(check_tfrac_thresh), 
+    censoring = function(param){
```

Comment:
Instead of having these check functions here, they should be added in an analogous way to check_aggregate, etc as separate functions in setup_helpers.R

Reply:
Actually, aren't some of these in the Stan parameters?